File: 0448-find-all-numbers-disappeared-in-an-array/0448-find-all-numbers-disappeared-in-an-array.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
const findDisappearedNumbers = nums => {
    const res = [];
    for (let i = 0; i < nums.length; i++) {
        const curr = Math.abs(nums[i]);
        if (nums[curr - 1] > 0) nums[curr - 1] *= -1;
    }
    console.log(nums)
    for (let i = 1; i <= nums.length; i++) {
        if (nums[i - 1] > 0) res.push(i);
    }
    return res;
};

File: 0448-find-all-numbers-disappeared-in-an-array/0448-find-all-numbers-disappeared-in-an-array.py
class Solution:
    def findDisappearedNumbers(self, nums: List[int]) -> List[int]:
        n = len(nums)
        numSet = set(nums)
        arr = []
        for i in range(1, n + 1):
            if i not in numSet:
                arr.append(i)
                
        return arr